### DISCUSSION TITLE
路由耗时过长

### DISCUSSION THREAD

**Discussion by ting772** [NervJS/taro#10080] [Q & A]
最近用小程序性能调试工具时突然注意到小程序路由耗时太长了。于是自己用原生简单的demo跑了下，发现路由就50ms左右，页面切换起来流畅。
    但是，线上小程序（使用taro3/react）居然路由平均耗时都到了200ms，有的时候500ms-800ms也是有可能的，切换页面感觉到明显的卡顿。

想问下，有路由耗时长由哪些因素导致的呢？跟项目大小有关吗？，需要怎么优化呢

**Comment by ting772** [comment]
算了无感了~，对taro这框架无语了。告诫各位写稍微大一点的项目，千万不要用taro，各种坑。
1.如上述问题，路由时长（navigateTo）验证增加，当项目庞大了后，体验极差，恶劣的情况下1000ms都有可能。
2.如果你的项目主包还要引入其它sdk，如融云，即构的话，那你的主包可用空间简直捉襟见肘，此时如果你组件库组织关系处理的不好，所有的样式，逻辑代码都会打到主包去，也就是业务你啥页没干，空间也没几百k剩下了。
3.不要来这里提讨论，没人看的，不活跃，问了也没回复

**Reply by itmanyong** [reply]
毕竟是模拟的web环境，所使用的API正常情况下都会经过中间层(就称之为Taro转换器吧)。而这中间有多少逻辑就不得而知，时间上绝对是会加长的。
项目包大的原因主要是因为Taro模拟的web环境包太大，里面包含了许多无用的适配，具体可见Taro3.4版本的功能计划部分。原生默认项目也就200kb+。Taro的初始包至少是3倍以上。
Taro构建的代码貌似对无用代码的拆除优化的很不好，总给人一种感觉第三方库有时候全量build了一样。
使用Taro开发一定记得主包最多一个主页面,其他页面全放在分包。防止后期坑死
